src/config/note.config.ts

```diff
@@ -6,14 +6,26 @@
 import { registerAs } from '@nestjs/config';
 import * as Joi from 'joi';
 
+import { DefaultAccessPermission } from './default-access-permission.enum';
 import { buildErrorMessage, parseOptionalNumber, toArrayConfig } from './utils';
 
 export interface NoteConfig {
   forbiddenNoteIds: string[];
   maxDocumentLength: number;
+  everyoneAccessAllowed: boolean;
```

Comment:
I don't get what the purpose of this option is.

Reply:
The idea was to have one option that completely disables guest access in HedgeDoc for example for restricted corporate instances.
The term "everyone" is quite misleading here and in other places, therefore we'd like to change it to "guest" in all occurences, but not in this PR.

Reply:
But then we shouldn't introduce this here. 
This PR is about default permissions

Reply:
Same is with create permissions.

Reply:
Please lets only focus on the default permission here. Then we can focus on the create permission AND the API logic for permissions at the same time

Reply:
The "May create" vars and the "everyoneAccessAllowed" var aren't even used yet. The default permissions are.

Reply:
imho everything after https://github.com/hedgedoc/hedgedoc/pull/2590/commits/51e955f2b22ad4e37a47605aadca29508d6e4586 except for https://github.com/hedgedoc/hedgedoc/pull/2590/commits/ab24964ac2bd8830de8b9ba8f61fe8a0890f49db doesn't belong into this PR because it's not working properly.